### PR TITLE
Dynamic Favicon

### PR DESCRIPTION
**Pull request by nmacianx**
## Related Issues

Related to https://github.com/descope/etc/issues/7047

## Description

💬 Tries to replace the favicon dynamically if the SSO app has one configured

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -71,6 +78,43 @@
...
+				if (favicon.includes('{ssoAppId}')) {
+					favicon = favicon.replace('{ssoAppId}', ssoAppId);
+				}
+
+				const validateFaviconUrl = (url: string) =>
+					new Promise((resolve) => {
+						const img = new Image();
+						img.onload = () => resolve(true);
+						img.onerror = () => resolve(false);
+						img.src = url;
```
[the remark this answers is not in the document]
Comment (dorsha) [reply src/App.tsx:97]: need to fix this

[the remark this answers is not in the document]
Comment (dorsha) [reply src/App.tsx:97]: and this

File: src/App.tsx [hunk 2]
```diff
@@ -71,6 +78,43 @@ const App = () => {
 		})
 	};
 
+	React.useEffect(() => {
+		const updateFavicon = async () => {
+			if (faviconUrl && ssoAppId && projectId) {
+				let favicon = faviconUrl;
+				if (favicon.includes('{projectId}')) {
+					favicon = favicon.replace('{projectId}', projectId);
```
Comment (dorsha) [inline src/App.tsx:86]: You can just do replace, you don't must to have the includes.

File: src/App.tsx [hunk 3]
```diff
@@ -71,6 +78,43 @@ const App = () => {
...
+				}
+				if (favicon.includes('{ssoAppId}')) {
+					favicon = favicon.replace('{ssoAppId}', ssoAppId);
```
Comment (dorsha) [inline src/App.tsx:89]: same

File: src/App.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (dorsha) [reply src/App.tsx:97]: I would fix by checking the ssoAppId and projectId are in the desired format.

**Comment by orius123** [comment]
Need to work against the dynamicDsk branch, not main

**Review by dorsha — APPROVED** [review]
See Eliya comment

**Pull request merged**